### PR TITLE
creditbureau_configuration

### PR DESCRIPTION
**Pull request by rrpawar96**
## Description
The Old PR #2318 which remains unmerged has been rebased and the previous comments have been addressed.

This is the front end of the configuration module for Adding CB (Credit Bureau) by its Alias Name and configuring CB with LoanProduct, 

The backend of this module which is already Merged: https://github.com/apache/fineract/pull/356

##Ticket: https://mifosforge.jira.com/browse/MIFOSX-2714

### CODE REVIEW

File: app/global-translations/locale-en.json [hunk 1]
```diff
@@ -1816,6 +1818,15 @@
   "#Headings": "..",
   "label.heading.interestDetails": "Interest Details",
   "label.heading.savingsDetails": "Savings Details",
+  "label.heading.Account":"Account ",
+  "label.heading.numberOfAccounts":"Number of  Accounts",
+  "label.heading.totalPastDues":"Total Past Dues",
+  "label.heading.numberOfPastDueAccounts":"Number Of Past Due Accounts",
+  "label.heading.totalWrittenOfAmount":"Total written of Amount",
+  "label.heading.institutionName":"Institution Name",
+  "label.heading.accountNumber":"Account Number",
```
Comment (nikpawar89) [inline app/global-translations/locale-en.json]: can you check, if this labels are used in any html pages. Only keep the labels which we are currently using.

File: app/scripts/controllers/configurations/ViewCreditBureauSummaryController.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+ */
+
+
+(function (module) {
+    mifosX.controllers = _.extend(module, {
+        ViewCreditBureauSummaryController: function ($scope, resourceFactory, $routeParams, location,route) {
+            $scope.Configs = [];
+            $scope.CBConfigs=[];
+            $scope.externalServicesType = $routeParams.externalServicesType;
+            //$scope.name = $routeParams.name;
```
Comment (nikpawar89) [inline app/scripts/controllers/configurations/ViewCreditBureauSummaryController.js]: take out the comments.

File: app/scripts/controllers/configurations/ViewCreditBureauSummaryController.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                else
+                {
+
+                    biD.innerHTML="enable";
+                }
+            }
+
+            $scope.changeCbStatus=function(cb_id,status)
+            {
+                //alert("status is"+status);
```
Comment (nikpawar89) [inline app/scripts/controllers/configurations/ViewCreditBureauSummaryController.js]: remove comments.

File: app/scripts/controllers/configurations/ViewCreditBureauSummaryController.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                var formData={};
+                formData.creditBureauId=cb_id;
+                formData.is_active=!status;
+
+                resourceFactory.creditBureauSummary.put(formData, function (data) {
+                    route.reload();
+                    /*location.path('/externalservicesCB/CreditBureau');*/
```
Comment (nikpawar89) [inline app/scripts/controllers/configurations/ViewCreditBureauSummaryController.js]: remove comments.

File: app/scripts/controllers/configurations/ViewCreditBureauSummaryController.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            }
+
+            $scope.changeMappingStatus=function (mappingid,status) {
+                var mappingUpdate={};
+                mappingUpdate.creditbureauLoanProductMappingId=mappingid;
+                mappingUpdate.is_active=!status;
+                resourceFactory.creditBureauMapping.put(mappingUpdate,function (data) {
+
+                    route.reload();
+                    /*location.path('/externalservicesCB/CreditBureau');*/
```
Comment (nikpawar89) [inline app/scripts/controllers/configurations/ViewCreditBureauSummaryController.js]: remove comments.

File: app/scripts/routes.js [hunk 6]
```diff
@@ -1010,9 +1010,19 @@
...
+                templateUrl: 'views/administration/CreditBureauSummary.html'
+            })
+             .when('/externalservicesCB/CreditBureau/addcb', {
+                 templateUrl: 'views/administration/addNewCreditBureau.html'
+            })
+            .when('/externalservicesCB/CreditBureau/mapcblp', {
+                templateUrl: 'views/administration/MapCreditBureauToLP.html'
+            })
             .otherwise({
-                templateUrl: "views/errors/404.html"
```
Comment (nikpawar89) [inline app/scripts/routes.js:1014]: is this needed.

File: app/views/administration/externalServices.html [hunk 7]
```diff
@@ -27,9 +27,7 @@ <h4 class="list-group-item-heading"><i class="fa fa-large fa fa-envelope"></i>&n
                         </div>
                     </div>
                 </div>
-            </div>
```
Comment (nikpawar89) [inline app/views/administration/externalServices.html:30]: did we break this.. i don't see card being closed </card>

**Review by nikpawar89 — CHANGES_REQUESTED** [review]
Please address above feedback. I hope this was tested with latest fineract code. After making these changes please test again.

File: app/global-translations/locale-en.json (hunk 1 above)
Comment (rrpawar96) [reply app/global-translations/locale-en.json]: Removed labels that are not used.

File: app/scripts/controllers/configurations/ViewCreditBureauSummaryController.js (hunk 2 above)
Comment (rrpawar96) [reply app/scripts/controllers/configurations/ViewCreditBureauSummaryController.js]: removed all the comments.

File: app/scripts/routes.js (hunk 6 above)
Comment (rrpawar96) [reply app/scripts/routes.js:1014]: comment has been addressed and reverted the changes.

File: app/views/administration/externalServices.html (hunk 7 above)
Comment (rrpawar96) [reply app/views/administration/externalServices.html:30]: this card has been closed at the bottom --line no. 62 for proper alignment.

**Comment by rrpawar96** [comment]
image
This is how it looks.(Admin > System > External Services > Credit Bureau Integration)

**Comment by nikpawar89** [comment]
Looks good to me, @edcable  can we assign someone with write access to merge this PR.

**Comment by edcable** [comment]
@gauravsaini03 @mbj36 @abhaychawla @botraunak Would either of you have time to merge this PR?

**Comment by nikpawar89** [comment]
@awasum this has been reviewed and ready to merge. If you have write access can you merge it? 
or can I have write access for time being until this project is completed?

**Comment by awasum** [comment]
Merging this base on @nikpawar89 's +1. Thanks

**Comment by awasum** [comment]
@nikpawar89  and @rrpawar96 . with this merged, Will you want to write on mailing list so people try this feature out? Will you like to write a small wiki page on how to use this from the end user's POV? cc @edcable @vorburger

**Comment by nikpawar89** [comment]
> @nikpawar89 and @rrpawar96 . with this merged, Will you want to write on mailing list so people try this feature out? Will you like to write a small wiki page on how to use this from the end user's POV? cc @edcable @vorburger

@awasum Detail documentation would be written on this entire feature. Thanks for merging this PR

**Comment by francisguchie** [comment]
> @nikpawar89 and @rrpawar96 . with this merged, Will you want to write on mailing list so people try this feature out? Will you like to write a small wiki page on how to use this from the end user's POV? cc @edcable @vorburger

I am going to try this out from a users / implementation point of view

**Pull request merged**